Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def resolve_url(url)
+    m = Utils.popen_read("curl", "-I", url).match /^Location: (.+)$/
+    m.nil? ? url : resolve_url(m.captures.first)
```

Comment:
A quick note here. It should be `m.nil? ? url : URI.join(url, resolve_url(m.captures.first)).to_s`